clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/security/KubernetesV2Credentials.java

```diff
@@ -115,15 +115,16 @@ public boolean getOnlySpinnakerManaged() {
 
   private final Path serviceAccountNamespacePath = Paths.get("/var/run/secrets/kubernetes.io/serviceaccount/namespace");
 
-  public boolean isValidKind(KubernetesKind kind) {
+  public Optional<String> checkIfInvalidKind(KubernetesKind kind) {
```

Comment:
The contract of this function has gotten a lot more complicated, changing from a simple `boolean` `isValid` to an `Optional` that all callers have to unwrap and negate.  We should think about how to support the need for additional info in `validateKind` without complicating the interface for all other callers.  As one possibility, `isValidKind` could keep the same contract, but another function `getInvalidReason` could be used in the one place where it's needed.